src/thread_private_spike_store.hpp

```diff
@@ -20,7 +20,7 @@ class thread_private_spike_store {
 public :
     using id_type = cell_gid_type;
     using time_type = Time;
-    using spike_type = spike<cell_member_type, time_type>;
+    using spike_type = basic_spike<cell_member_type, time_type>;
```

Comment:
Why not remove the parameterization on `Time` type for `thread_private_spike_store`, and just use `spike` instead of `basic_spike`?